pkg/datapath/loader/compile.go

```diff
@@ -288,6 +288,44 @@ func progCFlags(prog *progInfo, dir *directoryInfo) []string {
 	)
 }
 
+// We need to map all bpf/*.c files to ids (an id value doesn't matter, but
+// should be a unique non-zero __u8 number).  A build will fail if there is
+// a file found for which there is no a corresponding id.
+var sourceNameToId map[string]int = map[string]int{
+	"bpf_network.c":            1,
+	"bpf_sock.c":               2,
+	"bpf_host.c":               3,
+	"bpf_overlay.c":            4,
+	"bpf_xdp.c":                5,
+	"bpf_alignchecker.c":       6,
```

Comment:
I think we can exclude `bpf_alignchecker.c` and `cilium-probe-kernel-hz.c`, as both are not expected to push to the perf ringbuffer.

Reply:
I added all of the C files in `bpf/` so that I can automatically check if a file is here or not (see `make check-sources`). We have 256 IDs, so should be enough even if we add userspace files as well.